fastify.d.ts

```diff
@@ -1,26 +1,190 @@
+    logger?: boolean | FastifyLoggerOptions<RawServer> & PinoLoggerOptions | Logger,
+    serializerOpts?: FJSOptions | Record<string, unknown>,
+    serverFactory?: FastifyServerFactory<RawServer>,
+    caseSensitive?: boolean,
+    requestIdHeader?: string | false,
+    requestIdLogLabel?: string;
+    jsonShorthand?: boolean;
+    genReqId?: <RequestGeneric extends RequestGenericInterface = RequestGenericInterface, TypeProvider extends FastifyTypeProvider = FastifyTypeProviderDefault>(req: FastifyRequest<RequestGeneric, RawServer, RawRequestDefaultExpression<RawServer>, FastifySchema, TypeProvider>) => string,
+    trustProxy?: boolean | string | string[] | number | TrustProxyFunction,
+    querystringParser?: (str: string) => { [key: string]: unknown },
+    /**
+     * @deprecated Prefer using the `constraints.version` property
+     */
+    versioning?: {
+      storage(): {
+        get(version: string): string | null,
+        set(version: string, store: Function): void
+        del(version: string): void,
+        empty(): void
+      },
+      deriveVersion<Context>(req: Object, ctx?: Context): string // not a fan of using Object here. Also what is Context? Can either of these be better defined?
+    },
+    constraints?: {
+      [name: string]: ConstraintStrategy<FindMyWayVersion<RawServer>, unknown>,
+    },
+    schemaController?: {
+      bucket?: (parentSchemas?: unknown) => {
+        add(schema: unknown): FastifyInstance;
+        getSchema(schemaId: string): unknown;
+        getSchemas(): Record<string, unknown>;
+      };
+      compilersFactory?: {
+        buildValidator?: ValidatorCompiler;
+        buildSerializer?: SerializerCompiler;
+      };
+    };
+    return503OnClosing?: boolean,
+    ajv?: {
+      customOptions?: AjvOptions,
+      plugins?: (Function | [Function, unknown])[]
+    },
+    frameworkErrors?: <RequestGeneric extends RequestGenericInterface = RequestGenericInterface, TypeProvider extends FastifyTypeProvider = FastifyTypeProviderDefault, SchemaCompiler extends FastifySchema = FastifySchema>(
+      error: FastifyError,
+      req: FastifyRequest<RequestGeneric, RawServer, RawRequestDefaultExpression<RawServer>, FastifySchema, TypeProvider>,
+      res: FastifyReply<RawServer, RawRequestDefaultExpression<RawServer>, RawReplyDefaultExpression<RawServer>, RequestGeneric, FastifyContextConfig, SchemaCompiler, TypeProvider>
+    ) => void,
+    rewriteUrl?: (req: RawRequestDefaultExpression<RawServer>) => string,
+    schemaErrorFormatter?: (errors: FastifySchemaValidationError[], dataVar: string) => Error,
+    /**
+     * listener to error events emitted by client connections
+     */
+    clientErrorHandler?: (error: ConnectionError, socket: Socket) => void
+  }
+
+  export interface ValidationResult {
+    keyword: string;
+    instancePath: string;
+    schemaPath: string;
+    params: Record<string, string | string[]>;
+    message?: string;
+  }
+
+  /* Export additional types */
+  export type {
```

Comment:
I would clean up this export before landing.
In theory We can merge the export by using the same `namespace` in other file.

Reply:
You really want to use namespace for that?

Reply:
It is better than managing massive of import and export.
Using `multi file namespace` allow to export in other file than merge all those in one.

Reply:
It doesnt work as expected

Created a new PR with potential changes for this PR. 

https://github.com/fastify/fastify/pull/4491

Looking for feedback, on how we could potentially use namespace merging.

Reply:
I will take a look tomorrow.